### PR TITLE
Share jars between server and beeline to reduce binary size

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

Share jars between server and beeline to reduce binary size.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Review by yaooqinn — APPROVED** [review]

**Comment by ulysses-you** [comment]
seems a little hack ...

File: build/dist [hunk 1]
```diff
@@ -192,6 +192,14 @@ cp -r "$KYUUBI_HOME/kyuubi-assembly/target/scala-$SCALA_VERSION/jars/" "$DISTDIR
 # Copy kyuubi beeline jars
 cp "$KYUUBI_HOME"/kyuubi-hive-beeline/target/*.jar "$DISTDIR/beeline-jars/"
 
+# Share the jars between server and beeline to reduce binary size
+for jar in $(ls "$DISTDIR/jars/"); do
+  if [[ -f "$DISTDIR/beeline-jars/$jar" ]]; then
+    rm "$DISTDIR/beeline-jars/$jar"
+    ln -sn "$DISTDIR/jars/$jar" "$DISTDIR/beeline-jars/$jar"
```
Comment (pan3793) [inline build/dist:199]: Does the absolute path work if user untar to another folder?

Comment (turboFei) [reply build/dist]: nice catch

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging to master

**Pull request closed without merging**